File: lib/easy/reformat_date.dart
String reformatDate(String date) {
  List<String> months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
  List<String> parts = date.split(" ");
  int day = 0;
  for (final char in parts[0].codeUnits) {
    if (char >= 48 && char <= 57) {
      day = day * 10 + (char - 48);
    }
  }
  return "${parts[2]}-${(months.indexOf(parts[1]) + 1).toString().padLeft(2, "0")}-${day.toString().padLeft(2, "0")}";
}

File: src/easy/reformat-date.ts
const months = [
	"Jan",
	"Feb",
	"Mar",
	"Apr",
	"May",
	"Jun",
	"Jul",
	"Aug",
	"Sep",
	"Oct",
	"Nov",
	"Dec",
];
export const reformatDate = (date: string): string => {
	const [d, m, year] = date.split(" ");
	let day = 0;
	const month = months.indexOf(m) + 1;
	for (const char of d) {
		if (char.charCodeAt(0) >= 48 && char.charCodeAt(0) <= 57) {
			day = day * 10 + Number(char);
		}
	}
	return `${year}-${month.toString().padStart(2, "0")}-${day.toString().padStart(2, "0")}`;
};
